test/base/resources.go

```diff
@@ -59,8 +70,17 @@ func Channel(name, provisioner string) *eventingv1alpha1.Channel {
 	}
 }
 
+// KafkaChannel returns a KafkaChannel.
+func KafkaChannel(name string) *kafkamessagingv1alpha1.KafkaChannel {
```

Comment:
no namespace?

Reply:
maybe you can also pass the options as well? And create a spec? Look at inmemorychannel.go  inside pkg/reconciler/testing.  
Can some of the methods there for the different objects be reused? Or you prefer to have all the e2e infra separate?

Reply:
`namespace` gets automatically added when we use the namespaced `KafkaChannelInterface` to create it, so there is no need to add `namespace` here.
This method also applies to other resources.

Reply:
I would prefer to have them separated. Since a lot of functions like `WithInMemoryChannelDeploymentNotReady` and `WithInMemoryChannelDeploymentReady` are only for unit tests, having them available in e2e infra will be confusing.